packages/query-core/src/queryObserver.ts

```diff
@@ -698,11 +704,14 @@ export class QueryObserver<
         includedProps.add('error')
       }
 
-      return Object.keys(this.#currentResult).some((key) => {
+      return [...includedProps].some((key) => {
```

Comment:
Tiny optimization that also made debugging a bit easier

Reply:
this is a great improvement 🎉